homeassistant/components/plugwise/config_flow.py

```diff
@@ -108,6 +112,9 @@ async def async_step_user(self, user_input=None):
                 errors["base"] = "unknown"
             if not errors:
                 await self.async_set_unique_id(api.gateway_id)
+                await self.async_set_unique_id(
+                    api.smile_hostname or api.gateway_id, raise_on_progress=False
```

Comment:
Is the smile host name the same as the gateway id?

Reply:
For some older devices the smile_hostname isn't amongst the data so for cases where it would end up `None` we use the gateway_id (which is sort of an internal string/hash).

Reply:
Ok, so the same device will always use the same unique id.

Reply:
Why are we setting unique id again here if we're coming from the zeroconf step? Will we use the same unique id then as was set in the zeroconf step?

Reply:
Some people can't use zeroconf for some reason, so we are setting the same unique_id here.
Yes, it's the same as in the zeroconf-step.

Reply:
Ok.